### ISSUE TITLE
setDatetime or getInfo vs Automatic Mode

### ISSUE THREAD

**Issue by JacekDob** [maxnowack/node-eq3ble#5]
Hi,

After using getInfo or setDatetime(new Date()) for getting eq-3 status - automatic mode stops working or works unpredictable.

Looking on Android Application getting status is performed by full setdate time command not just '03'.
And on the other hand I see some differences especially year is not in full format but last 2 digits (modulo 100):

Set Time:
public SetTimeCommand() {
        this.commandData = new byte[7];
        this.commandData[0] = (byte) 3;
        GregorianCalendar calendar = new GregorianCalendar(TimeZone.getDefault());
        calendar.setTimeInMillis(System.currentTimeMillis());
        this.commandData[1] = (byte) (calendar.get(1) % 100);
        this.commandData[2] = (byte) (calendar.get(2) + 1);
        this.commandData[3] = (byte) calendar.get(5);
        this.commandData[4] = (byte) calendar.get(11);
        this.commandData[5] = (byte) calendar.get(12);
        this.commandData[6] = (byte) calendar.get(13);
    } 

Get Status:
    protected void onResume() {
        ...
            ((BLEApplication) getApp()).getBluetoothAPI().connectToGroup(this.groupID);
            ((BLEApplication) getApp()).getBluetoothAPI().sendCommandToGroup(this.groupID, new SetTimeCommand());
        ...
        updateGroupStateUI(group);
    }

Currently I have no fix for that.
As a workaround to get status from eq-3 and not disturbed automatic mode for getInfo I am using setAutomaticMode payload.

**Comment by JacekDob** [comment]
Got the fix:

* interface.js
```
setDatetime: (date) => {
  var b = Buffer.alloc(7);
  b[0] = 3;
  b[1] = (date.getFullYear() % 100);
  b[2] = (date.getMonth() + 1);
  b[3] = date.getDate();
  b[4] = date.getHours();
  b[5] = date.getMinutes();
  b[6] = date.getSeconds();
  return b;
}
```

* index.js
Proper getting info:
```
getInfo() {
  return this.writeAndGetNotification(payload.setDatetime(new Date())).then(info => parseInfo(info))
}
```

And parsing status for setTemperature (BT return status for this, maybe for more too):
```
setTemperature(temperature) {
  return this.writeAndGetNotification(payload.setTemperature(temperature)).then(info => parseInfo(info))
}
```

**Comment by maxnowack** [comment]
@JacekDob looks legit. would you be so kind and open a PR? :)

**Comment by gS32tom** [comment]
hey @maxnowack, because of this issue I can't use your homebridge-eq3ble plugin. It keeps changing the dates on the eq3. :( I already lost hours trying to implement @JacekDob's fork, but can't get it to work (node-noob here). I would greatly appreciate you pulling the changes and updating the package on npm. 👍

**Comment by ccoenen** [comment]
@maxnowack the work is probably all thre in his master branch (https://github.com/JacekDob/node-eq3ble/tree/de47ebf1d337327ad3e7b144b6a217f5a597a68b). I'd love for this to be part of the regular package.

**Comment by ccoenen** [comment]
@maxnowack pull request #6 contains the changeset. Could you merge it into this package? If not, how can I help to get this merged?

**Comment by maxnowack** [comment]
finally, #6 was merged and released in v1.2.0 😊
